File: Sorting/InsertionSort.js
class Solution {
  insert(arr, i) {
    // code here such that insertionSort() sorts arr[]
    for (let j = i; j > 0 && arr[j - 1] > arr[j]; j--) {
      let tmp = arr[j - 1];
      arr[j - 1] = arr[j];
      arr[j] = tmp;
    }
  }

  //Function to sort the array using insertion sort algorithm.
  insertionSort(arr, n) {
    //code here
    for (let i = 0; i < n; i++) {
      let sort = this.insert(arr, i);
    }
  }
}

File: Sorting/InsertionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void insertionSort(int arr[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = i; j > 0 && arr[j - 1] > arr[j]; j--)
        {
            int tmp = arr[j - 1];
            arr[j - 1] = arr[j];
            arr[j] = tmp;
        }
    }
}

int main()
{
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    insertionSort(arr, n);
    for (int i = 0; i < n; i++)
        cout << arr[i] << " ";
    return 0;
}